sw/device/lib/crypto/include/ecc.h

```diff
@@ -333,12 +336,19 @@ crypto_status_t otcrypto_ecdsa_keygen_async_start(ecc_curve_t *elliptic_curve);
  * `kCryptoStatusAsyncIncomplete` if the OTBN is busy or
  * `kCryptoStatusInternalError` if there is an error.
  *
+ * The caller must ensure that the `elliptic_curve` parameter matches the one
+ * that was previously passed to the corresponding `_start` function; a
+ * mismatch will cause inconsistencies. Similarly, the private key
+ * configuration must match the one originally passed to `_start`.
+ *
+ * @param elliptic_curve Pointer to the elliptic curve that is being used.
  * @param[out] private_key Pointer to the blinded private key (d) struct.
  * @param[out] public_key Pointer to the unblinded public key (Q) struct.
  * @return Result of asynchronous ECDSA keygen finalize operation.
  */
 crypto_status_t otcrypto_ecdsa_keygen_async_finalize(
-    crypto_blinded_key_t *private_key, ecc_public_key_t *public_key);
+    ecc_curve_t *elliptic_curve, crypto_blinded_key_t *private_key,
```

Comment:
Since `elliptic_curve` is marked as mutable, does it make sense to mark the documentation as `@param[inout]` and describe why it is mutable?  The same comment applies to all other occurrences of `elliptic_curve`.

Reply:
They shouldn't be mutable -- I'll fix that!